PatientWebApplication/HealthClinic.DAL/Service/RegistrationInPharmacyService.cs

```diff
@@ -25,17 +30,30 @@ public List<RegistrationInPharmacy> GetAll()
         public RegistrationInPharmacy Create(RegistrationInPharmacyDto dto)
         {
             RegistrationInPharmacy registration = RegistrationInPharmacyAdapter.RegistrationDtoToRegistration(dto);
-
-            return RegistrationInPharmacyRepository.Create(registration);
-        }
-
-        public RegistrationInPharmacy getApiKey(int pharmacyId)
+            if (isApiKeyUnique(registration.apiKey))  return RegistrationInPharmacyRepository.Create(registration); 
+             return null;
+        }
+        public bool isApiKeyUnique(String apiKey)
+        {
+            foreach(RegistrationInPharmacy registration in GetAll())
+            {
+                if (registration.apiKey.Equals(apiKey)) return false;
+            }
+            return true;
+        }
+        public List<RegistrationInPharmacy> GetAllForStub()
         {
-            return RegistrationInPharmacyRepository.getApiKey(pharmacyId);
+            return IRegistrationRepository.GetAll();
         }
-        public RegistrationInPharmacy getPharmacyId(String apiKey)
-        {
-            return RegistrationInPharmacyRepository.getPharmacyId(apiKey);
+
+        public RegistrationInPharmacy getPharmacyApiKey(String apiKey)
+        {
+            List<RegistrationInPharmacy> registrationInPharmacies = IRegistrationRepository.GetAll();

```

Comment:
Unnecessary instance of the List registrationInPharmacies. Do the same as in the 38 line of code.